src/Microsoft.Unity.Analyzers.Tests/ProtectedUnityMessageTests.cs

```diff
@@ -0,0 +1,159 @@
+/*--------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *-------------------------------------------------------------------------------------------*/
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Unity.Analyzers.Tests
+{
+	public class ProtectedUnityMessageTests : BaseCodeFixVerifierTest<ProtectedUnityMessageAnalyzer, ProtectedUnityMessageCodeFix>
+	{
+		[Fact]
+		public async Task FixPrivateUnityMessage()
+		{
+			const string test = @"
+using UnityEngine;
+
+class Camera : MonoBehaviour
```

Comment:
We should detect when a class is `sealed`, in this case we should not introduce protected members given we'll add `CS0628`